base/src/main/scala/typeclass/InvariantFunctor.scala

```diff
@@ -0,0 +1,10 @@
+package scalaz
+package typeclass
+
+trait InvariantFunctor[F[_]] {
+  def xmap[A, B](ma: F[A])(f: A => B)(g: B => A): F[B]
```

Comment:
@jdegoes I agree, imap is a better name, will change. 

Once we have Iso I believe we should also have an additional method that takes an Iso.  This is how we have it in scalaz 7.